anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java

```diff
@@ -282,19 +284,35 @@ private void checkMappedMetricData(MappedMetricData mappedMetricData) {
     void detectorLastUsedTimeSync(long currentTime) {
         long updateDurationInSeconds = (currentTime - detectorsLastUsedSyncedTillTime) / 1000;
 
-        if (updateDurationInSeconds <= 0 || detectorsLastUsedTimeToBeUpdatedSet.isEmpty()) {
+        if (updateDurationInSeconds <= 0 || detectorsLastUsedTimeToBeUpdatedQueue.isEmpty()) {
             return;
         }
+        updateDetectorsLastTimeUsed();
+        detectorsLastUsedSyncedTillTime = currentTime;
+    }
 
+    private void updateDetectorsLastTimeUsed() {
+        int detectorsLastUsedTimeQueueSize = detectorsLastUsedTimeToBeUpdatedQueue.size();
+        log.info("Detectors last used time queue size {}", detectorsLastUsedTimeQueueSize);
+
+        Set<UUID> detectorsLastUsedTimeToBeUpdatedSet = buildDetectorsLastTimeToBeUpdatedSet(detectorsLastUsedTimeQueueSize);
```

Comment:
I think we are looping twice here, once to convert the queue to HashSet, and another to consume the set and update the last access time. I think it can be modified to be single looping, with local HashSet variable named something like `processedUUIDSet` and perform checking and/or filling the set in the same loop that updating `lastAccessedTime` takes place